server/src/models/User.ts

```diff
@@ -4,12 +4,14 @@ class User {
   readonly isAdmin: boolean;
 
   vote: string | null;
+  hasVoted: boolean;
 
   constructor(name: string, id: string, isAdmin: boolean) {
     this.name = name;
     this.socket = id;
     this.isAdmin = isAdmin;
     this.vote = null;
+    this.hasVoted = false;

```

Comment:
Sugeruje skorzystać tutaj z gettera który zwraca true jeżeli this.vote nie jest nullem. W ten sposób nigdzie nie trzeba będzie ustawiać tej właściwości jawnie.

https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Functions/get